src/server/scripts/Spells/spell_item.cpp

```diff
@@ -333,6 +333,74 @@ class spell_item_blessing_of_ancient_kings : public SpellScriptLoader
         }
 };
 
+enum DeadlyPrecision
+{
+    SPELL_DEADLY_PRECISION = 71564
+};
+
+// 71564 - Deadly Precision
+class spell_item_deadly_precision : public SpellScriptLoader
+{
+    public:
+        spell_item_deadly_precision() : SpellScriptLoader("spell_item_deadly_precision") { }
+
+        class spell_item_deadly_precision_charm_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_item_deadly_precision_charm_AuraScript);
+
+            void HandleStackDrop(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+            {
+                PreventDefaultAction();
+                GetTarget()->RemoveAuraFromStack(GetId());
+            }
+
+            void Register() override
+            {
+                OnEffectProc += AuraEffectProcFn(spell_item_deadly_precision_charm_AuraScript::HandleStackDrop, EFFECT_0, SPELL_AURA_MOD_RATING);
+            }
+        };
+
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_item_deadly_precision_charm_AuraScript();
+        }
+};
+
+// 71563 - Deadly Precision Dummy
+class spell_item_deadly_precision_dummy : public SpellScriptLoader
+{
+    public:
+        spell_item_deadly_precision_dummy() : SpellScriptLoader("spell_item_deadly_precision_dummy") { }
+
+        class spell_item_deadly_precision_dummy_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_item_deadly_precision_dummy_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_DEADLY_PRECISION))
+                    return false;
+                return true;
+            }
+
+            void HandleDummy(SpellEffIndex /*effIndex*/)
+            {
+                SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(SPELL_DEADLY_PRECISION);
```

Comment:
As the spellInfo exists, because of Validate hook, use AssertSpellInfo so Coverity doesn't complain